tests/unit_tests/test_ustar_cp/test_cpdBootstrapUStarTh4Season20100901.py

```diff
@@ -0,0 +1,263 @@
+                                  generation of temperature values. Defaults to (-10, 30).
+        fNight (array-like or None, optional): Binary values indicating day (0) or 
+                                               night (1). If None, a random assignment 
+                                               is made. Defaults to None.
+
+    Returns:
+        tuple: A tuple containing:
+            - t (np.ndarray): Time vector of length `nt`.
+            - NEE (np.ndarray): Randomly generated Net Ecosystem Exchange values.
+            - uStar (np.ndarray): Randomly generated u* values.
+            - T (np.ndarray): Randomly generated temperature values.
+            - fNight (np.ndarray): Array indicating day (0) or night (1) conditions.
+    """
+    t = np.linspace(*tspan, nt)  # Generate time vector
+    NEE = np.piecewise(t, [t < 100, (t >= 100) & (t < 200), t >= 200],
+                       [lambda x: 2 * x + 1, lambda x: -x + 300, lambda x: 0.5 * x + 100])
+    uStar = np.random.uniform(*uStar_pars, size=nt)  # u* values between typical ranges
+    T = np.random.uniform(*T_pars, size=nt)  # Temperature values
+    if fNight == None:
+        fNight = np.random.choice([0, 1], size=nt)  # Randomly assign day/night
+    else:
+        fNight = np.resize(fNight, nt)
+    return t, NEE, uStar, T, fNight
+
+def test_cpdBootstrapUStarTh4Season20100901_basic(matlab_engine, mock_data):
+    t, NEE, uStar, T, fNight = mock_data
+    fPlot = 0
+    cSiteYr = "Site_2024"
+    nBoot = 10
+
+    # Convert mock inputs to MATLAB format
+    t_matlab = matlab.double(t.tolist())
+    NEE_matlab = matlab.double(NEE.tolist())
+    uStar_matlab = matlab.double(uStar.tolist())
+    T_matlab = matlab.double(T.tolist())
+    fNight_matlab = matlab.logical(fNight.tolist())
+
+    # Call MATLAB function
+    Cp2, Stats2, Cp3, Stats3 = matlab_engine.cpdBootstrapUStarTh4Season20100901(
+        t_matlab, NEE_matlab, uStar_matlab, T_matlab, fNight_matlab, fPlot, cSiteYr, nBoot, 1, nargout=4
+    )
+    Stats2 = json.loads(Stats2)
+    Stats3 = json.loads(Stats3)
+
+    # Assertions for output types
+    assert isinstance(Cp2, matlab.double), "Cp2 should be a MATLAB double array."
+    assert isinstance(Stats2, list), "Stats2 should be a list of MATLAB structs."
+    assert isinstance(Cp3, matlab.double), "Cp3 should be a MATLAB double array."
+    assert isinstance(Stats3, list), "Stats3 should be a list of MATLAB structs."
+
+    # Validate dimensions of the output arrays
+    assert len(Cp2) == 4, "Cp2 should have 4 seasons."
+    assert len(Cp3) == 4, "Cp3 should have 4 seasons."
+    assert len(Stats2) == 4, "Stats2 should have 4 entries for each season."
+    assert len(Stats3) == 4, "Stats3 should have 4 entries for each season."
+
+    # Check the structure of Stats2 and Stats3
+    for s2, s3 in zip(Stats2, Stats3):
+        for i in range(8):  # Assuming nStrataX = 8
+            for j in range(nBoot):
+                assert 'n' in s2[i][j]
```

Comment:
we could add a 4th loop here along the lines of:
for var in ['n', 'CP','fMax','p', 'b0' etc]:
    assert var in s2[I][j]
    assert var in s3[I][j]
    
Which would tidy up this long set of asserts

Reply:
Fixed